src/execution/isolate.cc

```diff
@@ -5384,9 +5389,16 @@ void Isolate::RunAllPromiseHooks(PromiseHookType type,
   }
 }
 
+extern void RecordReplayOnPromiseHook(Isolate* isolate, PromiseHookType type,
```

Comment:
Once moved to a separate file, we should just include the file. I'd love to avoid getting comfortable with `extern`ing all our dependencies. Or do we have a good reason to keep using `extern`?